core/http-auth/src/main/java/software/amazon/awssdk/http/auth/AwsV4HttpSigner.java

```diff
@@ -82,6 +83,47 @@ public interface AwsV4HttpSigner extends HttpSigner<AwsCredentialsIdentity> {
     SignerProperty<Boolean> NORMALIZE_PATH =
         SignerProperty.create(Boolean.class, "NormalizePath");
 
+    /**
+     * The location where auth-related data is inserted, as a result of signing.
+     * The valid choices are "Header" and "QueryString", where "Header" indicates that request
+     * headers are added, and "QueryString" indicates query-parameters are added.
+     * This property defaults to "Header".
+     */
+    SignerProperty<String> AUTH_LOCATION =
+        SignerProperty.create(String.class, "AuthLocation");
+
+    /**
+     * The duration for the request to be valid.
+     * This property defaults to the max valid duration (7 days).
+     * This is only used in the case of a pre-signing implementation.
+     */
+    SignerProperty<Duration> EXPIRATION_DURATION =
+        SignerProperty.create(Duration.class, "ExpirationDuration");
+
+    /**
+     * Whether to indicate that a payload is signed or not.
+     * This property defaults to true.
+     * This can be set false to disable payload signing.
+     */
+    SignerProperty<Boolean> PAYLOAD_SIGNING =
+        SignerProperty.create(Boolean.class, "PayloadSigning");
+
+    /**
+     * Whether to handle a request as an event-streaming or not.
+     * This property defaults to false.
+     * This is only used in the case of an implementation that supports event-streams.
+     */
+    SignerProperty<Boolean> EVENT_STREAMING =
```

Comment:
thinking aloud: I was thinking if this is really a Signer property. It feels it is really a property of the request / shape of the request than a configuration property of the signing process? An event streaming request needs special handling to sign the payload. By just saying PAYLOAD_SIGNING=true (and assuming there was another way to know that the request is event streaming) could enable the appropriate signing behavior. 

Then if another signer (e.g., sigv4a, or say another scheme in the future) supports signing event streams, it wouldn't be a new/distinct signer property in that Signer too.

> another way to know the request is event streaming?

To go this route, would need the SdkHttpRequest abstraction of SignRequest to be decorated/extended to include these aspects or separate member of SignRequest directly.

This would be a bigger change to the overall integration, as compared to SignerProperties that an AuthSchemeProvider can add based on `operation` AuthSchemeParameter. I'm not sure if this is worth pursuing, but wanted to share these thoughts here for consideration.

Reply:
Can we just look for the header `Content-Type: application/vnd.amazon.eventstream` to determine it's an event stream request? We can do something similar for chunked-encoding.

Reply:
Is that header a protocol level detail or defined by event streaming feature itself?

Reply:
It's specific to HTTP, but I believe it's true for all protocols. I don't know who "owns" it if we're going to be that abstract, but if it ever becomes not-true in the future, we could reasses then.

Reply:
Based on offline discussion, clarified that by protocol here I meant the serialization aspect too (e.g., restJson1 v/s awsJson1_0) and not specifically the transport protocol (HTTP or other). But we also clarified that the event streaming sigv4 spec is tied to `application/vnd.amazon.eventstream` so using that as a signal to decide how to sign is fine, as opposed to requiring an input SignerProperty to indicate the behavior. It kinda makes sigv4 support for event streaming tied to the amazon protocols, but sigv4 is also amazon specific, so that _coupling_ is fine. In a hypothetical world where someone uses sigv4 with their own protocol that uses something other than `application/vnd.amazon.eventstream`, they'd just have to provide their own sigv4 signer implementation too that support it, our default sigv4 signer implementation would only support the amazon protocols. This is fine :)